qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ProviderSteps.java

```diff
@@ -0,0 +1,103 @@
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.contrib.qaframework.automation;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.helper.RestClient;
+import org.openmrs.contrib.qaframework.helper.TestData;
+import org.openmrs.contrib.qaframework.page.AdministrationPage;
+import org.openmrs.contrib.qaframework.page.ManageProviderPage;
+import org.openmrs.contrib.qaframework.page.ProviderPage;
+
+public class ProviderSteps extends Steps {
+	
+    private TestData.PersonInfo person;
+    private String personUuid;
+    private String providerUuid;
+    private TestData.TestProvider provider;
+    private AdministrationPage administrationPage;
+    private ManageProviderPage manageProviderPage;
+    private ProviderPage providerPage;
+	
+    @Before(RunTest.HOOK.SELENIUM_PROVIDER)
+    public void setUp() {
+    	initiateWithLogin();
+    	person = TestData.generateRandomPerson();
+    	personUuid = TestData.createPerson(this.person);
+    	TestData.createPerson(this.person);
+    	provider = new TestData.TestProvider(person.uuid, person.uuid);
+    	providerUuid = provider.create();
+        providerUuid = new TestData.TestProvider(personUuid, personUuid).create();
+    }
+    
+    @After(RunTest.HOOK.SELENIUM_PROVIDER)
+    public void tearDown() {
+        RestClient.delete("provider/" + providerUuid, true);
+        RestClient.delete("person/" + personUuid, true);
+	quit();
+    }
+    
+    @Given("User clicks the system administration link on the home page")
+    public void launchSystemAdministrationPage() {
+    	administrationPage = homePage.goToAdministration();
+    }
+    
+    @When ("User clicks the manage Provider link on the administration page")
+    public void launchAdvancedAdministrationPage() {
+    	manageProviderPage = administrationPage.clickOnManageProviders();
+    }
+    
+    @And ("User clicks on add provider link")
+    public void clickOnAddProviderLink() {
+    	providerPage = manageProviderPage.clickOnAddProvider();
+    }
+    
+    @And ("User fills the provider form")
+    public void fillProviderForm() {
+        providerPage.setIdentifier(personUuid);
+        providerPage.setPerson(person.getName());
+        manageProviderPage = providerPage.clickOnSave();
+    }
+    
+    @And ("User searches for the created provider")
+    public void searchForProvider() {
+    	manageProviderPage.setProviderNameOrId(person.getName());
+    	providerPage = manageProviderPage.clickOnProvider(person.getName());
+    }
+	
+    @Then ("User deletes provider forever")
+    public void deleteProviderForever() {
```

Comment:
what do you think of `deleteProviderForever()` vs `deleteProvider()`?

Reply:
unfortunately ts already being used here    
 ```
@Then ("User deletes  a provider")
    public void deleteProvider() {
	providerPage.deleteForever();
    }
```